lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -0,0 +1,413 @@
+
+    def test_default_output_units(self):
+        """Test the function returns accumulations in the default units if no
+        units are explicitly set, where the default is metres."""
+
+        # Multiply the rates in mm/s by 60 to get accumulation over 1 minute
+        # and divide by 1000 to get into metres.
+        expected = self.cubes[0].copy(data=self.cubes[0].data * 60 / 1000)
+
+        plugin = Accumulation(accumulation_period=60)
+        result = plugin.process(self.cubes)
+
+        self.assertEqual(result[0].units, 'm')
+        self.assertArrayAlmostEqual(result[0].data, expected.data)
+
+    def test_default_altered_output_units(self):
+        """Test the function returns accumulations in the specified units if
+        they are explicitly set. Here the units are set to mm."""
+
+        # Multiply the rates in mm/s by 60 to get accumulation over 1 minute
+        expected = self.cubes[0].copy(data=self.cubes[0].data * 60)
+
+        plugin = Accumulation(accumulation_units='mm', accumulation_period=60)
+        result = plugin.process(self.cubes)
+
+        self.assertEqual(result[0].units, 'mm')
+        self.assertArrayAlmostEqual(result[0].data, expected.data)
+
+    def test_raises_exception_for_unevenly_spaced_cubes(self):
+        """Test function raises an exception if the input cubes are not
+        spaced equally in time."""
+
+        last_time = self.cubes[-1].coord('time').points
+        self.cubes[-1].coord('time').points = last_time + 60
+
+        msg = ("Accumulation is designed to work with rates "
+               "cubes at regualar time intervals.")
+        plugin = Accumulation(accumulation_period=120)
+
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin.process(self.cubes)
+
+    @ManageWarnings(ignored_messages=["The provided cubes result in a"],
+                    warning_types=[UserWarning])
+    def test_does_not_use_incomplete_period_data(self):
+        """Test function returns only 2 accumulation periods when a 4 minute
+        aggregation period is used with 10 minutes of input data. The trailing
+        2 cubes are insufficient to create another period and so are disgarded.
+        A warning is raised by the chunking function and has been tested above,
+        so is ignored here.
+        """
+
+        plugin = Accumulation(accumulation_period=240)
+        result = plugin.process(self.cubes)
+        self.assertEqual(len(result), 2)
+
+    def test_returns_expected_values_5_minutes(self):
+        """Test function returns the expected accumulations over a 5 minute
+        aggregation period. These are written out long hand to make the
+        comparison easy."""
+
+        expected_1st_row_t0 = np.array([
+            0.03, 0.06, 0.09, 0.12, 0.15, 0.21, 0.27, 0.33, 0.39, 0.45])
+        expected_2nd_row_t0 = np.array([
+            0.03, 0.06, 0.09, 0.12, 0.15,
+            np.inf, np.inf, np.inf, np.inf, np.inf])
```

Comment:
Why do we have `np.inf`?

Reply:
Because these values are masked and so don't actually get checked by this test. I could make them all -1, or something else. Making them zeroes would make it appear that some valid answer had been arrived at for these cells, whereas I want to indicate that they are masked. I've also added an explicit check that the mask is as expected to try and help clarify this.

Reply:
I think `np.nan` would be my preference here - just because infinity has an actual meaning (in the sense of potentially being the result of dividing by zero).  That was what confused me.

Reply:
Done.